### PR TITLE
Feature/dao

### PR DESCRIPTION
**Pull request by Gaaralmn**

### CODE REVIEW

File: daemon/test/client_test.go [hunk 1]
```diff
@@ -17,6 +23,52 @@ func (suite *ClientTestSuite) SetupTest() {
 	earningClient.SendData()
 }
 
+func (suite *ClientTestSuite) TestUpsert() {
+	fmt.Println("In TestUpsert")
+	//config.InitConfig()
+	//serviceConfig := config.GetConfig()
+	dbConfig := fmt.Sprintf("host=%s user=%s dbname=%s sslmode=disable password=%s",
+		"localhost", "postgres", "postgres", "docker")
```
Comment (Gaaralmn) [inline daemon/test/client_test.go:31]: @Michae-zHOU There are few errors there while I use config, so I hard code the DB connection here.

File: daemon/persistence/sampletask_dao.go [hunk 2]
```diff
@@ -2,52 +2,61 @@ package persistence
...
-	//}
+	prevReport := SampleTask{}
+	err := s.Db.Where("uid = ?", t.Uid).Last(&prevReport).Error
+	if errors.Is(err, gorm.ErrRecordNotFound) {
+		err :=s.Db.Create(&t).Error
+		if err != nil {
+			fmt.Println(err, t.ID)
+		}
+	} else {
+		s.Db.Save(&t)
```
Comment (singerdmx) [inline daemon/persistence/sampletask_dao.go:40]: only update column "content", not the whole thing

Comment (singerdmx) [reply daemon/persistence/sampletask_dao.go:40]: If you can load it from DB, update "content" field, and save the object loaded from DB, not the one passed into the function

Comment (Gaaralmn) [reply daemon/persistence/sampletask_dao.go:40]: I see I will adjust this sometime tomorrow

File: daemon/test/client_test.go (hunk 1 above)
Comment (Michae-zHOU) [reply daemon/test/client_test.go:31]: It is ok for now. We can adjust testing later.

File: daemon/persistence/sampletask_dao.go (hunk 2 above)
Comment (Michae-zHOU) [reply daemon/persistence/sampletask_dao.go:40]: Add duedate, duetime change for the case of IPO deadline changed

File: daemon/test/client_test.go (hunk 1 above)
Comment (Gaaralmn) [reply daemon/test/client_test.go:31]: Cool

File: daemon/persistence/sampletask_dao.go [hunk 3]
```diff
@@ -2,52 +2,68 @@ package persistence
...
+		if err != nil {
+			fmt.Println(err, t.ID)
+		}
+	} else {
+		//Find the existing SampleTask Entity based on ID
+		var existSt SampleTask
+		s.Db.First(&existSt, t.ID)
+		//Update the SampleTask only if content change
+		if existSt.Content != t.Content {
+			t.Content = existSt.Content
```
Comment (singerdmx) [inline daemon/persistence/sampletask_dao.go:45]: DO this to DueDate, availableBefore, DueTime as well

**Pull request merged**